### ISSUE TITLE
POST Application Error

### ISSUE THREAD

**Issue by ygizhitsa** [Azure-Samples/blockchain#212]
### This issue is for a: (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

- Create service principal
- Get Bearer token 
- make sure access I granted 

Have followed above steps
- Successfully call Applications GET method - returns list of the ABW applications 
`$response = Invoke-RestMethod -Method Get -Uri $applicationsUrl -ContentType "application/json" -Headers $headers
$response |  ConvertTo-Json`
![image](https://user-images.githubusercontent.com/16059989/68324675-db718b00-0095-11ea-9a33-62a2678787f8.png)

Execute  POST in order to automate BasicProvenence application creation 
`$response = Invoke-RestMethod -Method POST -Uri $applicationsUrl -ContentType "multipart/form-data"   -Headers $headers -InFile .\BasicProvenance.json -Verbose

Error:
![image](https://user-images.githubusercontent.com/16059989/68324575-a9602900-0095-11ea-96a3-b218cb45134d.png)

`

**Comment by ygizhitsa** [comment]
Fiddler version of the call

![image](https://user-images.githubusercontent.com/16059989/68327695-d6173f00-009b-11ea-81e1-cb5866a9e870.png)

**Comment by ygizhitsa** [comment]
Also, in relation  to the following "Implementation Notes
Creates a new blockchain application. This method can only be performed by users who are Workbench administrators."

I have double check service principal is Administrator for ABW

**Comment by ygizhitsa** [comment]
See

2019-11-08T14:12:04.859160767Z: [INFO]  [14:12:04 INF] {Protocol="HTTP/1.1", Method="POST", ContentType="multipart/form-data", ContentLength=4573, Scheme="http", Host="azdobc-j7sksw-api.azurewebsites.net", PathBase="", Path="/api/v1/applications", QueryString="", EventId={Id=1}, SourceContext="Microsoft.AspNetCore.Hosting.Internal.WebHost", RequestId="0HLQR8CBLVOF7:00000001", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Request starting HTTP/1.1 POST http://azdobc-j7sksw-api.azurewebsites.net/api/v1/applications multipart/form-data 4573
2019-11-08T14:12:04.860460208Z: [INFO]  [14:12:04 INF] {EventId={Id=2}, SourceContext="Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler", RequestId="0HLQR8CBLVOF7:00000001", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} **Successfully validated the token.**
2019-11-08T14:12:04.861245433Z: [INFO]  [14:12:04 INF] {EventId={Id=8}, SourceContext="Microsoft.AspNetCore.Authentication.JwtBearer.JwtBearerHandler", RequestId="0HLQR8CBLVOF7:00000001", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} AuthenticationScheme: Bearer was successfully authenticated.
2019-11-08T14:12:04.861852653Z: [INFO]  [14:12:04 INF] {SourceContext="data-access", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Execute [dbo].[GetUserByExternalId] with externalId = 0ec0b976-7be6-4836-8448-377e4d53d7b8
2019-11-08T14:12:04.862392470Z: [INFO]  [14:12:04 INF] {SourceContext="data-access", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Executing sproc [dbo].[GetUserByExternalId]
2019-11-08T14:12:04.868326460Z: [INFO]  [14:12:04 INF] {SourceContext="data-access", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Execute [dbo].[GetUserChainMappings] with id = 2
2019-11-08T14:12:04.868896578Z: [INFO]  [14:12:04 INF] {SourceContext="data-access", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Executing sproc [dbo].[GetUserChainMappings]
2019-11-08T14:12:04.873250117Z: [INFO]  [14:12:04 INF] {EventId={Id=1}, SourceContext="Microsoft.AspNetCore.Authorization.DefaultAuthorizationService", ActionId="0235c93e-f19b-4e3e-ba51-79dff7e9397a", ActionName="Microsoft.AppBuilder.API.Controllers.ApplicationsController.ApplicationsPostAsync (Microsoft.AppBuilder.API)", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} **Authorization was successful for user:** 0ec0b976-7be6-4836-8448-377e4d53d7b8.
2019-11-08T14:12:04.874017442Z: [INFO]  [14:12:04 INF] {EventId={Id=1}, SourceContext="Microsoft.AspNetCore.Authorization.DefaultAuthorizationService", ActionId="0235c93e-f19b-4e3e-ba51-79dff7e9397a", ActionName="Microsoft.AppBuilder.API.Controllers.ApplicationsController.ApplicationsPostAsync (Microsoft.AppBuilder.API)", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Authorization was successful for user: 0ec0b976-7be6-4836-8448-377e4d53d7b8.
2019-11-08T14:12:04.875360285Z: [INFO]  [14:12:04 INF] {EventId={Id=2}, SourceContext="Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker", ActionId="0235c93e-f19b-4e3e-ba51-79dff7e9397a", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Executed action Microsoft.AppBuilder.API.Controllers.ApplicationsController.ApplicationsPostAsync (Microsoft.AppBuilder.API) in 2.0839ms
2019-11-08T14:12:04.876434219Z: [INFO]  [14:12:04 ERR] {SourceContext="AppBuilderErrorHandler", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} **Unknown Exception encountered**
2019-11-08T14:12:04.878364681Z: [INFO]  [14:12:04 ERR] {SourceContext="AppBuilderErrorHandler", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} **Returning Internal Server Error (500)**
2019-11-08T14:12:04.878842596Z: [INFO]  **_System.IO.InvalidDataException: Missing content-type boundary._**
2019-11-08T14:12:04.878858597Z: [INFO]     at Microsoft.AspNetCore.Http.Features.FormFeature.GetBoundary(MediaTypeHeaderValue contentType, Int32 lengthLimit)
2019-11-08T14:12:04.878914798Z: [INFO]     at Microsoft.AspNetCore.Http.Features.FormFeature.InnerReadFormAsync(CancellationToken cancellationToken)
2019-11-08T14:12:04.879952332Z: [INFO]     at Microsoft.AspNetCore.Mvc.ModelBinding.FormValueProviderFactory.AddValueProviderAsync(ValueProviderFactoryContext context)
2019-11-08T14:12:04.879968732Z: [INFO]     at Microsoft.AspNetCore.Mvc.ModelBinding.CompositeValueProvider.CreateAsync(ActionContext actionContext, IList`1 factories)
2019-11-08T14:12:04.880703656Z: [INFO]     at Microsoft.AspNetCore.Mvc.ModelBinding.CompositeValueProvider.CreateAsync(ControllerContext controllerContext)
2019-11-08T14:12:04.880719356Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ControllerBinderDelegateProvider.<>c__DisplayClass0_0.<<CreateBinderDelegate>g__Bind|0>d.MoveNext()
2019-11-08T14:12:04.880725756Z: [INFO]  --- End of stack trace from previous location where exception was thrown ---
2019-11-08T14:12:04.880730557Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
2019-11-08T14:12:04.880735357Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
2019-11-08T14:12:04.882245205Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
2019-11-08T14:12:04.882260505Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
2019-11-08T14:12:04.882266606Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
2019-11-08T14:12:04.882271606Z: [INFO]     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
2019-11-08T14:12:04.882276406Z: [INFO]     at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
2019-11-08T14:12:04.883810155Z: [INFO]     at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.Invoke(HttpContext context)
2019-11-08T14:12:04.883826256Z: [INFO]     at Microsoft.AppBuilder.API.UserInfoMiddleware.InvokeAsync(HttpContext context) in /appbuilder.api/UserInfoMiddleware.cs:line 89
2019-11-08T14:12:04.883831856Z: [INFO]     at Microsoft.AppBuilder.API.AppBuilderErrorHandler.Invoke(HttpContext context) in /appbuilder.api/AppBuilderErrorHandler.cs:line 34
2019-11-08T14:12:04.885696615Z: [INFO]  [14:12:04 INF] {ElapsedMilliseconds=26.739, StatusCode=500, ContentType="application/json", EventId={Id=2}, SourceContext="Microsoft.AspNetCore.Hosting.Internal.WebHost", RequestId="147e4e4b-cfcb-429e-8b81-3360ffea3a62", RequestPath="/api/v1/applications", ServiceName="appbuilder.api"} Request finished in 26.739ms 500 application/json

**Comment by ygizhitsa** [comment]
Anyone from @Microsoft available?

**Comment by ramyapradhan** [comment]
Hello @ygizhitsa,

I have posted a solution to the issue that you have brought up in issue #210, and I think this and the other thread may be related.  Please let me know if the solution in #210 helps.

Thanks,
Ramya

**Comment by ramyapradhan** [comment]
Hello @ygizhitsa, please feel free to re-open this thread if this issue is not solved from the solution to #210 
Thanks,
Ramya
